lib/modules/manager/gitlabci/extract.ts

```diff
@@ -1,18 +1,38 @@
 import is from '@sindresorhus/is';
 import { logger } from '../../../logger';
 import { readLocalFile } from '../../../util/fs';
+import { regEx } from '../../../util/regex';
 import { trimLeadingSlash } from '../../../util/url';
 import { parseSingleYaml } from '../../../util/yaml';
+import { GitlabTagsDatasource } from '../../datasource/gitlab-tags';
 import type {
   ExtractConfig,
   PackageDependency,
   PackageFile,
   PackageFileContent,
 } from '../types';
-import { isGitlabIncludeLocal } from './common';
-import type { GitlabPipeline, Image, Job, Services } from './types';
+import {
+  filterIncludeFromGitlabPipeline,
+  isGitlabIncludeComponent,
+  isGitlabIncludeLocal,
+  isNonEmptyObject,
+} from './common';
+import type {
+  GitlabInclude,
+  GitlabIncludeComponent,
+  GitlabPipeline,
+  Image,
+  Job,
+  Services,
+} from './types';
 import { getGitlabDep, replaceReferenceTags } from './utils';
 
+// See https://docs.gitlab.com/ee/ci/components/index.html#use-a-component
+const componentReferenceRegex = regEx(
+  /(?<fqdn>[^/]+)\/(?<projectPath>.+)\/(.+)@(?<specificVersion>.+)/,
```

Comment:
```suggestion
  /(?<fqdn>[^/]+)\/(?<projectPath>.+)\/(?:.+)@(?<specificVersion>.+)/,
```
just a small improvement